### PR TITLE
Prod release v0.4.0

### PR DESCRIPTION
**Pull request by emanuelgaspar**
New method: `GetSupportedQuoteCurrencies`

New feature: `GetAllPrices` will normalise prices to the requested quote currencies even if they are not supported directly by the `CoinGecko` API.

**Breaking change** in `GetAllPrices`, now returns a custom object of type `MultiplePrices` instead of a dictionary of dictionaries.

`CoinGecko` Open API definition updated:
 - explicit boolean flags when applicable
 - `supported_vs_currencies` returns an array of strings

### CODE REVIEW

File: src/Trakx.CoinGecko.ApiClient/CoinGeckoClient.cs [hunk 1]
```diff
@@ -47,183 +38,226 @@ public class CoinGeckoClient : ICoinGeckoClient
...
-            .ConfigureAwait(false);
+        var supportedCurrencies = await GetSupportedQuoteCurrencies(cancellationToken);
 
-        var fxRate = await GetUsdFxRate(quoteCurrencyId, date);
+        (var baseIds, var quoteIds) = GetIdsForPriceQuery(ids, vsCurrencies, supportedCurrencies);
 
-        if (fullData.Content.Market_data == null)
-            return null;
+        var response = await _simpleClient
+            .PriceAsync(baseIds, quoteIds, cancellationToken: cancellationToken)
```
Comment (monsieurleberre) [inline src/Trakx.CoinGecko.ApiClient/CoinGeckoClient.cs:129]: Here did you test with a fairly big list of base currencies? I think I remember seeing it failing with an error like 
`input string too long, should be less that 1024 characters` or something like that

**Review by monsieurleberre — APPROVED** [review]
OK on principle, just one small question

File: src/Trakx.CoinGecko.ApiClient/CoinGeckoClient.cs (hunk 1 above)
Comment (emanuelgaspar) [reply src/Trakx.CoinGecko.ApiClient/CoinGeckoClient.cs:129]: https://trakx.atlassian.net/browse/TKX-860

I ran a test directly against the API, via `Postman`.
It worked OK for 1099 chars.

Repeating my comment from JIRA:
The current PR will be submitted without a “pagination” of symbols when getting prices, I will add one after if that’s OK.

So if you agree, my plan is:
- have the current (already meaty) PR merged and released ASAP
- return to the `market-data` PR, to get `CoinGecko` prices in bulk
- once the `CoinGecko` usage is tamed, I return to `coingecko-api-client` and add support for a limitless list of tokens.

Comment (emanuelgaspar) [reply src/Trakx.CoinGecko.ApiClient/CoinGeckoClient.cs:129]: Test added in #104

**Pull request merged**